swayfmt/src/items/item_impl/mod.rs

```diff
@@ -38,14 +38,27 @@ impl Format for ItemImpl {
             where_clause.format(formatted_code, formatter)?;
             formatter.shape.code_line.update_where_clause(true);
         }
-        Self::open_curly_brace(formatted_code, formatter)?;
+
         let contents = self.contents.get();
-        for item in contents.iter() {
-            write!(formatted_code, "{}", formatter.indent_str()?,)?;
-            item.format(formatted_code, formatter)?;
-            writeln!(formatted_code)?;
+        if contents.is_empty() {
```

Comment:
Let's add a couple test cases for this fix under `swayfmt/src/items/item_impl/tests.rs`